charts/datadog/templates/_container-system-probe.yaml

```diff
@@ -81,7 +81,7 @@
       mountPropagation: {{ .Values.datadog.hostVolumeMountPropagation }}
       readOnly: true
 {{- end }}
-{{- if and (or .Values.datadog.systemProbe.enableTCPQueueLength .Values.datadog.systemProbe.enableOOMKill) .Values.datadog.systemProbe.enableDefaultKernelHeadersPaths }}
+{{- if and (or (eq (include "runtime-compilation-enabled" .) "true") .Values.datadog.systemProbe.enableCORE) .Values.datadog.systemProbe.enableDefaultKernelHeadersPaths }}
```

Comment:
Why does CO-RE need kernel headers? It shouldn't need this mount.

Reply:
While we primarily expect default kernel BTFs to be in `/sys/kernel/btf/vmlinux` (which already gets mounted), they can sometimes  be found in `/lib/modules`. It's a bit confusing to lump this in with the kernel header configurations though, so I could separate this into a separate statement for clarity if necessary.

Reply:
oh actually - I just double checked the default BTF locations and realized the default BTFs can be found in `/lib/modules` and `/usr/lib`, not `/usr/src`. So it doesn't make sense to lump these together - let me fix that.

Reply:
([source](https://github.com/cilium/ebpf/blob/9da42c2fd67e8761067fcd9c27eb24b23acdb51c/btf/btf.go#L274-L319))

Reply:
There is a much larger list of searched locations https://github.com/cilium/ebpf/blob/master/btf/btf.go#L301-L307
I don't think we can get away with mounting `/boot` though.

I'm tempted to not have any mounts for CO-RE because:
1. They can specify a custom path if necessary
2. It should be rare that the kernel BTF isn't in the default location and we don't have a bundled BTF file

Simplifying the security concerns and mounting requirements for non-helm setups is another benefit.

Reply:
Don't worry, I wasn't planning on mounting `/boot` :) Just `/lib/modules` and `/usr/lib/` (though I could make it `/usr/lib/modules` + `/usr/lib/debug` instead to minimize the mounts).

Is it a security concern if we're already mounting similar paths for kernel header downloading? I agree that this is probably a rare case, which means I don't think we should bother worrying about it for non-helm setups, but if we can add support for that case here, I don't see why we wouldn't.

Reply:
So the only case where those files would be used, is if they had installed the package with the debug kernel **and it had BTF embedded** right? That seems sufficiently rare to not worry about, especially if there is a workaround.